### PR TITLE
Attribute defined outside init

### PR DESCRIPTION
**Pull request by fabianegli**
There were some issues with the tests not using pytest.raises for checking for exceptions raised and there were attributes defined outside init in the PipelineSync.

There was also this pattern to use open to create files which has now been changed to the more idiomatic `touch()` method from pathlib.Path.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: tests/test_sync.py [hunk 1]
```diff
@@ -20,81 +23,68 @@ class TestModules(unittest.TestCase):
...
         self.create_obj.init_pipeline()
 
     def tearDown(self):
-        if os.path.exists(self.tmp_dir):
+        if self.tmp_dir.exists():
             shutil.rmtree(self.tmp_dir)
 
     @with_temporary_folder
-    def test_inspect_sync_dir_notgit(self, tmp_dir):
+    def test_ensure_git_reop_notgit(self, tmp_dir):
```
Comment (mashehu) [inline tests/test_sync.py:38]:
```suggestion
    def test_ensure_git_repo_notgit(self, tmp_dir):
```

File: nf_core/sync.py [hunk 2]
```diff
@@ -24,16 +25,24 @@
...
 
-    pass
-
 
 class PullRequestException(Exception):
     """Exception raised when there was an error creating a Pull-Request on GitHub.com"""
 
-    pass
 
+def ensure_git_reop(directory: Path):
```
Comment (mashehu) [inline nf_core/sync.py:33]:
```suggestion
def ensure_git_repo(directory: Path):
```

File: nf_core/sync.py [hunk 3]
```diff
@@ -64,7 +73,8 @@ def __init__(
     ):
         """Initialise syncing object"""
 
-        self.pipeline_dir = os.path.abspath(pipeline_dir)
+        self.pipeline_dir = Path(pipeline_dir).absolute()
+        self.repo = ensure_git_reop(self.pipeline_dir)
```
Comment (mashehu) [inline nf_core/sync.py:77]:
```suggestion
        self.repo = ensure_git_repo(self.pipeline_dir)
```

Comment (fabianegli) [reply nf_core/sync.py:77]: The suggestions seem not to change anything.

@mashehu Was there a different version here when the changes were proposed?

Comment (fabianegli) [reply nf_core/sync.py:77]: and by version i mean a different function name...

**Pull request closed without merging**